android/engine/src/test/java/org/smartregister/fhircore/engine/data/local/DefaultRepositoryTest.kt

```diff
@@ -806,6 +812,50 @@ class DefaultRepositoryTest : RobolectricTest() {
     Assert.assertEquals(CarePlan.CarePlanStatus.COMPLETED, carePlan.status)
   }
 
+  @Test
+  fun testFilterRelatedResourcesShouldReturnTrueIfExpressionIsProvided() {
+    val resourceConfig =
+      ResourceConfig(
+        resource = ResourceType.Task,
+        filterFhirPathExpressions = listOf(KeyValueConfig("Task.status", "ready"))
+      )
+    val task =
+      Task().apply {
+        id = "37793d31-def5-40bd-a2e3-fdaf5a0ddc53"
+        status = Task.TaskStatus.READY
+      }
+    val result = defaultRepository.filterRelatedResource(task, resourceConfig)
+    Assert.assertTrue(result)
+  }
+
+  @Test
+  fun testFilterRelatedResourcesShouldReturnFalseIfExpressionIsProvided() {
```

Comment:
Update test name to be more descriptive e.g testFilterRelatedResourcesShouldReturnFalseIfProvidedExpressionEvaluatesToFalse